Libs/MRML/Core/vtkMRMLSegmentationNode.cxx

```diff
@@ -1186,4 +1186,34 @@ void vtkMRMLSegmentationNode::SetLabelmapConversionColorTableNodeID(const char*
 vtkMRMLColorTableNode* vtkMRMLSegmentationNode::GetLabelmapConversionColorTableNode()
 {
   return vtkMRMLColorTableNode::SafeDownCast(this->GetNodeReference(this->GetLabelmapConversionColorTableNodeReferenceRole()));
-}
\ No newline at end of file
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLSegmentationNode::OnNodeReferenceAdded(vtkMRMLNodeReference* reference)
+{
+  this->Superclass::OnNodeReferenceAdded(reference);
+  if (std::string(reference->GetReferenceRole()) == this->GetReferenceImageGeometryReferenceRole())
+    {
+    this->InvokeEvent(vtkMRMLSegmentationNode::ReferenceImageGeometryChangedEvent, reference->GetReferencedNode());
```

Comment:
Is there a specific reason not to use InvokeCustomModifiedEvent (which can be blocked by Start/EndModify)?